lib/tests/formulae_dependents.rb

```diff
@@ -29,15 +29,11 @@ def dependent_formulae!(formula_name, args:)
           dependents_for_formula(formula, formula_name, args: args)
 
         source_dependents.each do |dependent|
-          bottled = with_env(HOMEBREW_SKIP_OR_LATER_BOTTLES: "1") do
-            dependent.bottled?
-          end
-
-          # TODO: A better thing to do here would be to attempt a source build
-          #       of unbottled dependents, but reverse the failure condition; i.e.
-          #       a successful source build is a failure, but a failed source build
-          #       is a pass to enable detection of unbottled formulae that should be bottled.
-          next unless bottled
+          # TODO: Work out how to use this code to identify
+          #       dependents that can be bottled. See
+          #       https://github.com/Homebrew/homebrew-test-bot/pull/678
+          next unless bottled?(dependent, no_older_versions: true)
+          next if bottled?(dependent, :all) && dependent.deps.any? { |d| !bottled?(d.to_formula) }
```

Comment:
Unsure if I should be passing `no_older_versions: true` here too, as in
```suggestion
          next if bottled?(dependent, :all) && dependent.deps.any? { |d| !bottled?(d.to_formula, no_older_versions: true) }
```

Reply:
Yeh, I reckon probably. Better to have missing tests here than false negatives/positives. Same below.

Reply:
Done, however, I also enabled testing of formulae with `:all` bottles whose dependency was built on the current OS earlier in the `test-bot` run.

For example, some dependents of `mono` have an `:all` bottle, but `mono` has no Linux bottle. We probably want to check those `mono` dependents if/when `mono` is bottled on Linux in a `test-bot` run.

Reply:
Works for me 👍🏻